index.js

```diff
@@ -1,8 +1,261 @@
+  let coordinateX2 = parseInt(prompt("choose your `x` coordinate to strike. between 0-3".));
+  let coordinateY2 = parseInt(prompt("choose your `y` coordinate to strike. between 0-3".));
+
+  // Step 3b: Determine if the Player Sunk their Opponent's Ship
+  // Using the x and y coordinates from the previous step,
+  // check the opponent's board to see if the space (array element) at those indices is a ship (is equal to 1).
+  if (player1.gameBoard[coordinateX2][coordinateY2] === 1) {
+    // If yes, "sink the ship" (change the value from 1 to 0)
+    player1.gameBoard[coordinateX2][coordinateY2] = 0;
+    // decrement the opponent's ship count,
+    player1.shipCount -= 1;
+    console.log(player1.gameBoard);
+    // and show an alert that says "Hit!"
+    alert(`Hit!`)
+  } else if (player1.shipCount === 0) {
+    alert(`Congratulations ${currentUser} you WIN!!!`)
+    break;
+  } else {
+    alert(`You missed!`);
+    currentUser = player1.name;
+  }
+}
+}
+
+
+
+
+return `The winner is ${currentUser}!Congratulations!`
+}
+
+*/
+
+
+let player1 = {
+  name: prompt("Player1, what's you name?"),
+  shipCount: 4,
+  // 2-dimensional array with 4 rows and 4 columns, filled with zeros
+  gameBoard: [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ],
+};
+let player2 = {
+  name: prompt("Player2, what's you name?"),
+  shipCount: 4,
+  gameBoard: [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ],
+};
+
+for (var i = 0; i < 4; i++) {
+  console.log("count+" + player1.shipCount);
+  if (player1.shipCount > 0) {
+    var xPosP1 = Math.floor(Math.random() * 4);
+    var yPosP1 = Math.floor(Math.random() * 4);
+    while (player1.gameBoard[xPosP1][yPosP1] === 1) {
+      xPosP1 = Math.floor(Math.random() * 4);
+      yPosP1 = Math.floor(Math.random() * 4);
+    }
+
+    // creating gameBoard
+    player1.gameBoard[xPosP1][yPosP1] = 1;
+    //delete the shipcount, does not need it there
+  }
+  if (player2.shipCount > 0) {
+    var xPosP2 = Math.floor(Math.random() * 4);
+    var yPosP2 = Math.floor(Math.random() * 4);
+    // if that space has ship we do a while loop to assign a random number until we get an empty spot
+    while (player2.gameBoard[xPosP2][yPosP2] === 1) {
+      xPosP2 = Math.floor(Math.random() * 4);
+      yPosP2 = Math.floor(Math.random() * 4);
+    }
+    // player 2 has its ship on this position
+    player2.gameBoard[xPosP2][yPosP2] = 1;
+  }
+}
+
+
+// this function asks for coordinates for player1.
+
+function actionPlayer1() {
+  // we ask for a guess.
+  let x = prompt("player1 x:");
+  let y = prompt("player1 y:");
+  let xNum = parseInt(x);
+  let yNum = parseInt(y);
+
+  //
+
+  if (player2.gameBoard[xNum][yNum] === 1) {
+
+    // in here the players2's ship got hit
+    player2.gameBoard[xNum][yNum] = 0;
+    player2.shipCount -= 1;
+    console.log("player2 " + player2.shipCount);
+
+    alert("HIT! "
```

Comment:
in line 200 and 218 alert("HIT! "
      player1.name " has " + player1.shipCount + " ships left.") before and after players name + sign is missing